### ISSUE TITLE
Exposing the gradients from the Model class

### ISSUE THREAD

**Issue by shayzweig** [fchollet/keras#3080]
Hi,
I saw several issues opened regarding how to get the gradients of the network wrt. the loss function. Since I encountered with the same problem I thought I would share my solution and propose a permanent fix.

What I did is just expose the gradients as a function from the model class. I created a different function similar to _make_train_function that returns a theano function that computes the gradients (using get_gradients) then I created a function called gradients _of_batch that is similar to train_on_batch but instead of performing the updates it only returns the gradients.

Does this seem like a robust solution to the issue? it sounds to me like there is a growing demand for analysis of the gradients and this could be a good way to answer it. 

Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [ x] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [ x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [ x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by ChristianThomae** [comment]
Would be interesting if this could be connected to the metrics subsystem, so things like gradient norms etc. could be logged.

**Comment by shayzweig** [comment]
In my current implementation the gradients are returned as a dictionary similar to the one that is returned by get_weights. Therefore it is rather easy to compute the norm...
